src/crimson/os/seastore/seastore.cc

```diff
@@ -2583,6 +2617,29 @@ SeaStore::Shard::omaptree_get_values(
   });
 }
 
+SeaStore::Shard::omaptree_iterate_ret
+SeaStore::Shard::omaptree_iterate(
+  Transaction& t,
+  omap_root_t&& root,
+  ObjectStore::omap_iter_seek_t &start_from,
+  std::function<ObjectStore::omap_iter_ret_t(std::string_view, std::string_view)> &f)
+{
+  LOG_PREFIX(SeaStoreS::omaptree_iterate);
+  auto type = root.get_type();
+  DEBUGT("{} seek_position={} ", t, type,
+             start_from.seek_position);
```

Comment:
> save key to start_from.seek_position in callback function f, then next try will start from this seek_position for seastore repeat.

I don't see where `start_from.seek_position` is **saved/updated** in one transaction, which shall be used in the next retry transaction upon conflict.

Can you elaborate ?

Reply:
callback(f) will update start_from.seek_position.